pkg/ecdsa/ecdsa.go

```diff
@@ -0,0 +1,63 @@
+package ecdsa
+
+import (
+	"context"
+	"fmt"
+	"github.com/ipfs/go-log"
+	"github.com/keep-network/keep-common/pkg/persistence"
+	"github.com/keep-network/keep-core/pkg/net"
+	"github.com/keep-network/keep-core/pkg/sortition"
+)
+
+var logger = log.Logger("keep-ecdsa")
+
+// Initialize kicks off the ECDSA by initializing internal state, ensuring
+// preconditions like staking are met, and then kicking off the internal ECDSA
+// implementation. Returns an error if this failed, otherwise returns a handle
+// to the ECDSA node.
+func Initialize(
```

Comment:
I do not think it makes sense to separate `ecdsa` and `tbtc` nodes. I would only have a `tbtc` node.

The client needs to observe a lot of TBTC events and compose Bitcoin transactions according to TBTC v2 protocol rules. It's not like in v1 where the `keep-ecdsa` client was just reacting to DKG events and signature requests from the chain (before we realized there is a bunch of applications-specific actions). In v2 we have the Keep client supporting Random Beacon and TBTC protocols.

I would even go further and exposed only one chain instance: `TbtcChain` that integrates `WalletRegistry` and `Bridge` contracts.

We can have `ecdsa` package for DKG and signing logic - including retries - but I do not think we should have a separate node or chain.

Reply:
> I do not think it makes sense to separate ecdsa and tbtc nodes. I would only have a tbtc node.

Of course, that makes sense as well. Just to clarify, my reasoning was:

The ECDSA contracts living in `solidity/ecdsa` module are a separate being that, in theory, can be used by various applications, not only TBTC. This is why I proposed a similar design of client-side packages. This way we can keep pure ECDSA actions logically separated from the TBTC-specific actions and achieve a more flexible design for the future. 

> I would even go further and exposed only one chain instance: TbtcChain that integrates WalletRegistry and Bridge contracts.

This is what I planned to do anyway, even with the proposed design.

Reply:
See https://github.com/keep-network/keep-core/pull/3093/commits/50bea06c534eccd8cc3e393954d68f595eb703bd